apps/meteor/client/views/admin/rooms/RoomsTable.tsx

```diff
@@ -219,33 +189,67 @@ const RoomsTable = ({ reload }: { reload: MutableRefObject<() => void> }): React
 								</Box>
 							</Box>
 						</Box>
-					</Table.Cell>
-					<Table.Cell>
+					</GenericTableCell>
+					<GenericTableCell>
 						<Box color='hint' fontScale='p2m' style={style}>
 							{t(getRoomType(room))}
 						</Box>
 						<Box mi='x4' />
-					</Table.Cell>
-					<Table.Cell style={style}>{usersCount}</Table.Cell>
-					{mediaQuery && <Table.Cell style={style}>{msgs}</Table.Cell>}
-					{mediaQuery && <Table.Cell style={style}>{isDefault ? t('True') : t('False')}</Table.Cell>}
-					{mediaQuery && <Table.Cell style={style}>{featured ? t('True') : t('False')}</Table.Cell>}
-				</Table.Row>
+					</GenericTableCell>
+					<GenericTableCell style={style}>{usersCount}</GenericTableCell>
+					{mediaQuery && <GenericTableCell style={style}>{msgs}</GenericTableCell>}
+					{mediaQuery && <GenericTableCell style={style}>{isDefault ? t('True') : t('False')}</GenericTableCell>}
+					{mediaQuery && <GenericTableCell style={style}>{featured ? t('True') : t('False')}</GenericTableCell>}
+				</GenericTableRow>
 			);
 		},
 		[mediaQuery, onClick, t],
 	);
 
 	return (
-		<GenericTable
-			header={header}
-			renderRow={renderRow}
-			results={displayData}
-			total={data?.total}
-			params={params}
-			setParams={setParams}
-			renderFilter={({ onChange, ...props }): ReactElement => <FilterByTypeAndText setFilter={onChange} {...props} />}
-		/>
+		<>
+			<FilterByTypeAndText setFilter={setRoomFilter} />
+			{isLoading && (
+				<GenericTable>
+					<GenericTableHeader>{headers}</GenericTableHeader>
+					<GenericTableBody>
+						<GenericTableLoadingTable headerCells={6} />
+					</GenericTableBody>
+				</GenericTable>
+			)}
+			{isSuccess && data && data?.rooms.length > 0 && (
+				<>
+					<GenericTable>
+						<GenericTableHeader>{headers}</GenericTableHeader>
+						<GenericTableBody>{isSuccess && data?.rooms.map((room) => renderRow(room))}</GenericTableBody>
+					</GenericTable>
+					<Pagination
+						divider
+						current={current}
+						itemsPerPage={itemsPerPage}
+						count={data?.total || 0}
+						onSetItemsPerPage={setItemsPerPage}
+						onSetCurrent={setCurrent}
+						{...paginationProps}
+					/>
+				</>
+			)}
+			{isSuccess && data && data.rooms.length === 0 && (
+				<States>
+					<StatesIcon name='magnifier' />
+					<StatesTitle>{t('No_results_found')}</StatesTitle>
+				</States>
+			)}
+			{isError && (
+				<States>
+					<StatesIcon name='warning' variation='danger' />
+					<StatesTitle>{t('Something_went_wrong')}</StatesTitle>
+					<StatesActions>
+						<StatesAction onClick={() => refetch()}>{t('Reload_page')}</StatesAction>
```

Comment:
be careful when making major changes unrelated to the correctness or purpose of the PR

In addition, this component is deprecated, we have a 2.0 version so it doesn't seem to make much sense to maintain

The "generic table" was not as generic as we thought, that's why we created V2, as you can see V1 can use v2 (since this one is considerably more generic)

Reply:
Got it! Thanks @ggazzo